saleor/graphql/translations/mutations/utils.py

```diff
@@ -139,11 +165,42 @@ def perform_mutation(  # type: ignore[override]
         return cls(**{cls._meta.return_field_name: instance})
 
 
+class BaseTranslateMutationWithSlug(BaseTranslateMutation):
+    slugable_field = "name"
+
+    class Meta:
+        abstract = True
+
+    @classmethod
+    def pre_update_or_create(cls, instance, input_data, language_code):
+        translation_instance = instance.translations.filter(
+            language_code=language_code
+        ).first()
+
+        if translation_instance is None:
+            translation_instance = instance.translations.model()
+
+        validate_slug_already_exists(
+            instance, translation_instance, input_data, language_code
+        )
+
+        language_code_lookup = {"language_code": language_code}
+        input = validate_slug_and_generate_if_needed(
+            translation_instance,
+            cls.slugable_field,
```

Comment:
Not sure if this proper flow:

- for non existing translations (CREATE):
    -  when input.slug provided, save it - 👍 
    - **when input.slug is not provided, but input.name is provided, use `name` to create slug** - 👎 
    - when input.slug is not provided and input.name is not provided, save slug as `None`  - 👍 
- for  existing translations (UPDATE):
    - when input.slug is provided, save it  - 👍 
    - when translation.slug==None, and input.name is provided: do not change anything in slug field - 👍 

So right now, we're implicitly using `input.name` to generate the `slug` value, but only in specific condition.
Adding translation is an optional replacement for the specific field. So saving `name` should only save the translation for `name`. Saving `slug` should only save the `slug` field. 
This should simplify your logic, I think.

Reply:
I agree, I wasn't sure with this what is the best way to go. My arguments for this implementation were:
- When creating "regular" slugs, by default it is derived from name/title, so it would be consistent to do the same for translated slugs. The problem is that translated slug is optional which makes the thing a bit awkward
- I didn't want to duplicate validation code or modify existing slug too much, given that this is quite core feature to existing users and I am new to the codebase (which is probably a bad argument)  

I see your point and I will adjust accordingly